rc/windowing/zellij.kak

```diff
@@ -0,0 +1,69 @@
+# https://zellij.dev/
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+
+provide-module zellij %{
+
+# ensure we're running under zellij
+evaluate-commands %sh{
+    [ -z "${client_opt_windowing_modules}" ] || [ -n "$ZELLIJ" ] || echo 'fail zellij not detected'
+}
+
+declare-option -hidden -docstring %{zellij run options (
+    -c, --close-on-exit            Close the pane immediately when its command exits
+        --cwd <CWD>                Change the working directory of the new pane
+    -d, --direction <DIRECTION>    Direction to open the new pane in
+    -f, --floating                 Open the new pane in floating mode
+    -h, --help                     Print help information
+    -n, --name <NAME>              Name of the new pane
+    -s, --start-suspended          Start the command suspended, only running after you first presses ENTER)}\
+    str zellij_run_options ""
+declare-option -hidden -docstring %{zellij run command} str zellij_run_cmd ""
+
+define-command -hidden -params 0 zellij-run %{
+    nop %sh{
+    zellij run ${kak_opt_zellij_run_options} -- $SHELL -i -c "${kak_opt_zellij_run_cmd}" < /dev/null > /dev/null 2>&1 &
+    }
+    set global zellij_run_options ""
+    set global zellij_run_cmd ""
+}
+
+define-command -hidden -params 1.. zellij-terminal-impl %{
+    set global zellij_run_cmd "%arg{@}"
+    zellij-run
+}
+complete-command zellij-terminal-impl shell
+
+define-command zellij-terminal-vertical -params 1.. -docstring '
+zellij-terminal-vertical <program> [<arguments>]: create a new terminal as a zellij pane
+The current pane is split into two, top and bottom
+The program passed as argument will be executed in the new terminal' \
+%{
+    set global zellij_run_options "--direction down"
+    set global zellij_run_cmd "%arg{@}"
+    zellij-run
+}
+complete-command zellij-terminal-vertical shell
+
+define-command zellij-terminal-horizontal -params 1.. -docstring '
+zellij-terminal-horizontal <program> [<arguments>]: create a new terminal as a zellij pane
+The current pane is split into two, left and right
+The program passed as argument will be executed in the new terminal' \
+%{
+    set global zellij_run_options "--direction right"
+    set global zellij_run_cmd "%arg{@}"
+    zellij-run
+}
+complete-command zellij-terminal-horizontal shell
+
+define-command zellij-focus -params 0 -docstring '
+zellij-focus: focus a panel' \
+%{ nop %sh{
+    zellij action switch-mode pane
+}}
+
+## The default behaviour for the `new` command is to open an horizontal pane in a zellij session
+alias global focus zellij-focus
+alias global terminal zellij-terminal-impl
+
```

Comment:
`zellij-terminal-impl` does not seem to work correctly with `connect.kak` which probably means that arguments are not forwarded correctly. 

Also, `connect.kak` requires `popup` to be defined but considering that we are not providing it by convention across all `windowing` implementations so far (and it's up to a user to setup a hook correctly to bind extra aliases), this is something that can be left for a different discussion

Reply:
I didn't find `connect.kak`, could you point me where it is or what's the issue?

Reply:
Fount it, it's a plugin, will try it.

Reply:
I gave up this approach, it's very hard to pass arguments and make them in correct quotes.